src/debugging/DockerServerReadyAction.ts

```diff
@@ -213,11 +215,30 @@ class DockerLogsTracker extends vscode.Disposable {
             context.errorHandling.suppressDisplay = true;
             context.errorHandling.rethrow = false;
 
-            this.logStream = await ext.dockerClient.getContainerLogs(context, this.containerName) as LogStream;
+            this.logStream = new stream.PassThrough();
 
             this.logStream.on('data', (data) => {
                 this.detector.detectPattern(data.toString());
             });
+
+            // Don't wait
+            void runWithDefaultShell(
+                client => client.logsForContainer({ container: this.containerName, follow: true }),
+                ext.runtimeManager,
+                {
+                    cancellationToken: this.cts.token,
+                    stdOutPipe: this.logStream,
+                }
+            ).then(
+                () => {
+                    // Do nothing on fulfilled
+                },
+                () => {
+                    // Do nothing on reject
+                    // Needed because the usual termination path is for a cancellation error to be thrown,
+                    // because disposal of this object cancels the token and thus the process
```

Comment:
I'll add some details. In short, cancelling the token kills the process which makes `runWithDefaultShell` throw a `CommandCancelledError`. If you `void` something and it throws, it will bubble up to the extension host process. The host process eats it, but in the "Show Running Extensions" view it leaves warnings, and also spits them to the host process' console.